cmd/podman/images/scp.go

```diff
@@ -0,0 +1,354 @@
+				scpOpts.ToRemote = true
+				scpOpts.SourceImageName = args[1]
+			} else if strings.Contains(args[1], "::") { // both remote clients
+				scpOpts.FromRemote = true
+				scpOpts.ToRemote = true
+				if remoteArgLength(args[0], 1) == 0 { // is save->load w/ one image name or save-load w/ two image names
+					cliConnections = append(cliConnections, args[0])
+					cliConnections = append(cliConnections, args[1])
+				} else if remoteArgLength(args[0], 1) > 0 && remoteArgLength(args[1], 1) > 0 {
+					//in the future, this function could, instead of rejecting renames, also set a DestImageName field
+					return nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+				} else { // else its a load save (order of args)
+					cliConnections = append(cliConnections, args[1])
+					cliConnections = append(cliConnections, args[0])
+				}
+			} else {
+				//in the future, this function could, instead of rejecting renames, also set a DestImageName field
+				return nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+			}
+		} else if strings.Contains(args[1], "::") { // if we are given image host::
+			if remoteArgLength(args[1], 1) > 0 {
+				//in the future, this function could, instead of rejecting renames, also set a DestImageName field
+				return nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+			}
+			err := validateImageName(args[0])
+			if err != nil {
+				return nil, err
+			}
+			scpOpts.SourceImageName = args[0]
+			scpOpts.ToRemote = true
+			cliConnections = append(cliConnections, args[1])
+		} else {
+			//in the future, this function could, instead of rejecting renames, also set a DestImageName field
+			return nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify an image rename")
+		}
+	}
+	for i, val := range cliConnections {
+		splitEnv := strings.SplitN(val, "::", 2)
+		scpOpts.Connections = append(scpOpts.Connections, splitEnv[0])
+		if len(splitEnv[1]) != 0 {
+			err := validateImageName(splitEnv[1])
+			if err != nil {
+				return nil, err
+			}
+			scpOpts.SourceImageName = splitEnv[1]
+			//TODO: actually use the new name given by the user
+		}
+		conn, found := cfg.Engine.ServiceDestinations[scpOpts.Connections[i]]
+		if found {
+			scpOpts.Iden = append(scpOpts.Iden, conn.Identity)
+			url := conn.URI
+			urlt, err := urlP.Parse(url) // create an actual url to pass to exec command
+			if err != nil {
+				return nil, err
+			}
+			scpOpts.URI = append(scpOpts.URI, urlt)
+		} else { // no match, warn user and do a manual connection.
+			url := "ssh://" + scpOpts.Connections[i]
+			urlt, err := urlP.Parse(url) // create an actual url to pass to exec command
+			if err != nil {
+				return nil, err
+			}
+			scpOpts.URI = append(scpOpts.URI, urlt)
+			scpOpts.Iden = append(scpOpts.Iden, "")
+			logrus.Warnf("Unknown connection name given. Please use system connection add to specify the default remote socket location")
+		}
```

Comment:
Non-blocking: can be simplified further to
```go
var (urlString, iden…)
if found {
  urlString = conn.URI
  iden = conn.Identity
} else {
  urlString = "ssh://" + …
  iden = ""
}
url = urlP.Parse(urlString)
if url.User.Username() == "" { … }

scpOpts.URI = append(…, url)
scpOpts.Iden = append(…, iden)
```
which would be a bit less repetitive, and most importantly it would be easier to read follow that the goal of the loop is to create the two arrays.